libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/AztecEditorFragment.java

```diff
@@ -1008,6 +1008,35 @@ public void onMediaDeleted(AztecAttributes aztecAttributes) {
         if (!TextUtils.isEmpty(localMediaId)) {
             mEditorFragmentListener.onMediaDeleted(localMediaId);
         }
+        removeCaptionFromDeletedMedia(localMediaId);
```

Comment:
I think this should be inside the `if` clause above, so it doesn't get called when `localMediaId` is empty.

Reply:
Related to the above, this fix isn't working for me. I added a photo from the WP media library (not one from the device) then added a caption, but when I deleted it the caption remained. Debugging here showed the `localMediaId` was null.

Reply:
You are right, this is not going to work reliably in all cases. We can't rely on `localMediaId`, or any id devised from image attributes for this matter. It's also possible to have multiple images with the same id, and we will have no idea what caption to remove. Looks like this is another thing that has to be done on the Aztec side - we need to listen for removed image spans and delete any captions attached to them.